src/Yaapii.Atoms/Map/Sorted.cs

```diff
@@ -0,0 +1,178 @@
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, Func<KeyValuePair<string, Value>, KeyValuePair<string, Value>, int> compare)
+            : this(pairs, new ComparerFunc<KeyValuePair<string, Value>>(compare))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given key comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<string> cmp)
+            : this(pairs, new KeyComparer<string, Value>(cmp))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<KeyValuePair<string, Value>> cmp)
+            : base(
+                () =>
+                {
+                    var items = new List<KeyValuePair<string, Value>>(pairs);
+                    items.Sort(cmp);
+                    return new Map.Of<string, Value>(items);
+                },
+                false
+            )
+        { }
+    }
+
+    public sealed class Sorted : Map.Envelope
+    {
+        /// <summary>
+        /// Sorts the given map with the default comperator of the key
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs)
+            : this(pairs, Comparer<string>.Default)
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given compare function
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="compare">Function to compare two elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs, Func<KeyValuePair<string, string>, KeyValuePair<string, string>, int> compare)
+            : this(pairs, new ComparerFunc<KeyValuePair<string, string>>(compare))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given key comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs, IComparer<string> cmp)
+            : this(pairs, new KeyComparer<string, string>(cmp))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs, IComparer<KeyValuePair<string, string>> cmp)
```

Comment:
here too

Accept another Dictionary instead and please find a way to only access values if you need to (meaning if you want to use a Value comparer - so that we prevent building values that we do not need to build)

Reply:
A KeyValuePair / Dictinary contains types or function which are already created but no methods are used except the self implemented IComparer / compare function does this.

I see your point of ensuring that for future refactoring so I added a test in 4078206d addressing this.